File: backend/src/main/java/com/concessionaria/backend/model/Cliente.java
package com.concessionaria.backend.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "clientes")
public class Cliente {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_cliente")
    private Integer idCliente;

    @Column(nullable = false, length = 120)
    private String nome;

    @Column(nullable = false, length = 150)
    private String email;

    @Column(nullable = false, length = 20)
    private String telefone;

    @Column(nullable = false, length = 150)
    private String rua;

    @Column(nullable = false, length = 10)
    private String numero;

    @Column(nullable = false, length = 8)
    private String cep;

    public Cliente() {
    }

    public Cliente(String nome, String email, String telefone,
                   String rua, String numero, String cep) {
        this.nome = nome;
        this.email = email;
        this.telefone = telefone;
        this.rua = rua;
        this.numero = numero;
        this.cep = cep;
    }

    public Integer getIdCliente() {
        return idCliente;
    }

    public void setIdCliente(Integer idCliente) {
        this.idCliente = idCliente;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getTelefone() {
        return telefone;
    }

    public void setTelefone(String telefone) {
        this.telefone = telefone;
    }

    public String getRua() {
        return rua;
    }

    public void setRua(String rua) {
        this.rua = rua;
    }

    public String getNumero() {
        return numero;
    }

    public void setNumero(String numero) {
        this.numero = numero;
    }

    public String getCep() {
        return cep;
    }

    public void setCep(String cep) {
        this.cep = cep;
    }
}

File: frontend/src/app/core/models/cliente.ts
export type TipoCliente = 'FISICA' | 'JURIDICA';

export interface ClienteCadastroRequest {
  nome: string;
  email: string;
  telefone: string;
  rua: string;
  numero: string;
  cep: string;
  tipo: TipoCliente;
  cpf: string | null;
  cnpj: string | null;
}

export interface ClienteAtualizacaoRequest {
  nome: string;
  email: string;
  telefone: string;
  rua: string;
  numero: string;
  cep: string;
  tipo: TipoCliente;
  cpf: string | null;
  cnpj: string | null;
}

export interface ClienteResponse {
  idCliente: number;
  nome: string;
  email: string;
  telefone: string;
  rua: string;
  numero: string;
  cep: string;
  tipo: TipoCliente;
  cpf: string | null;
  cnpj: string | null;
}